### ISSUE TITLE
Unexpected behavior with Estimator warm start

### ISSUE THREAD

**Issue by dlsmith** [tensorflow/tensorflow#18696]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS X 10.12.6
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7.0
- **Python version**: 2.7.14
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Describe the problem
I've run into two issues when using a warm start with the `Estimator` API (please let me know if you'd like me to file these as separate issues):
1. If checkpoints exist in the `model_dir` then the warm start is overridden when the session is created.
2. The warm start is performed on every `train` call, which means warm starting can't be used in the context of `train_and_evaluate` (or in a manual loop, unless reinitializing the `Estimator`).

I'd be willing (pending corporate CLA) to submit a patch if you can clarify the desired behavior. I would personally expect that:
1. Checkpoint restore is skipped on session creation if a warm start was performed.
2. The warm start settings would be cleared after the first `train` call. (However, if this is the desired behavior, looking longer term it seems like warm start settings shouldn't be a property of the `Estimator` instance.)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 21 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ispirmustafa** [comment]
This is expected behavior. We define warm start as a starting point. If there is a checkpoint in model-dir, that means training is already started. If there is no checkpoint that means it's first training.

**Comment by neheller** [comment]
What about the second point? I agree with @dlsmith that it seems more natural to only "warm start" on the first `train` call.

**Comment by ispirmustafa** [comment]
warm-start says how you initialize your model. Same as defining initialization when you create a variable. It's part of the model definition. But used only if there isn't any checkpoint. Same as initialization.

**Comment by heyitsguay** [comment]
Has there been any update on part (2) here? I'm running into this problem as well. Despite @ispirmustafa 's comment, the behavior i'm seeing is that warm start reoccurs with each new call to the model_fn in TRAIN mode.  The warm-start log material shows up repeatedly, and my model's training loss follows a similar trajectory with each epoch.
